logging/logger.go

```diff
@@ -0,0 +1,121 @@
+// MIT License. Copyright (c) 2020 CQFN
+// https://github.com/cqfn/degitx/blob/master/LICENSE
+
+// Package logging contains degitx logging API, mapping and configuring internal implementation.
+// zap is current implementation
+package logging
+
+import (
+	"os"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type LogLevel int8
+
+const (
+	Debug LogLevel = 0
```

Comment:
@stepanvalyavskiy you can use `iota` syntax for levels https://github.com/golang/go/wiki/Iota

Reply:
@stepanvalyavskiy alos, you can use https://godoc.org/golang.org/x/tools/cmd/stringer standard tool to automatically generate `String()` method for log levels:
```go
//go:generate stringer -type=LogLevel
type LogLevel int8
```

Reply:
> @stepanvalyavskiy you can use `iota` syntax for levels https://github.com/golang/go/wiki/Iota

I thought we have a linter to avoid iota, at least I've read an article that speaks against iota syntax. Can't find now.

Reply:
> @stepanvalyavskiy alos, you can use https://godoc.org/golang.org/x/tools/cmd/stringer standard tool to automatically generate `String()` method for log levels:
> 
> ```go
> //go:generate stringer -type=LogLevel
> type LogLevel int8
> ```

ok, done